### DISCUSSION TITLE
Octokit auto-complete/suggestions not working

### DISCUSSION THREAD

**Discussion by veronikaEve** [octokit/octokit.js#2263] [Q&A]
I was watching this [video](https://www.youtube.com/watch?v=H1TupRbbjSI&ab_channel=LearnWithJason), and at the beginning, at 14:56, there is a huge list of suggested requests presented for Jason. Gregor (@gr2m) mentioned this is something they have been working on so does this should suggest it should work out of the box?

I dont get these suggestions. :( Do I have to configure something for it to work? 
I have the most useful intellisense settings on already, so basic editor suggestions work. Just this fancy Octokit one doesn't :/ 

Does it work for anyone else, or do I need some sort of configuration/setting?

Thanks in advance ✨

**Comment by wolfy1339** [comment]
The auto-complete is still there, it just got broken on newer versions of Typescript.

We pushed a fix so route auto-completion would work, however, that ended up breaking the options auto-completion.

There's an issue somewhere for this

**Reply by gr2m** [reply]
A temporary workaround is to wrap the route with `()`, e.g. `octokit.request((""))`, for some reason that triggers the route autocomplete again with the latest TypeScript version. It's really annoying, we hope that the next TS version will fix this bug

**Reply by veronikaEve** [reply]
So I was wondering if it was something to do with typescript, so then I changed my file to a simple .js. file. That did improve some of the suggestions but I still don't get all of it :(
![image](https://user-images.githubusercontent.com/42213048/181110517-76ba2bbd-5c64-4e30-bc66-5081f4b28ace.png)

I don't know what I'm doing wrong T.T

(Thanks for the swift reply)

**Reply by oscard0m** [reply]
Here is a TypeScript playground with the autocomplete problems: [link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzlApgRwK4oM7wL5wBmUEIcA5AAIQDGMEA1sDAPSqY4zkDcAUPyxZxAMuRwAghhg1SYADYoYKOABEA8gFEAynAByqgCpwA7tAa92WXAAoARAHF1+mwEoBQwKDk4ydPDzFx02xzdEsYK1sHJ2dXXlBIWERVOkZmAmJSOBtaeiYYGz5eaQA7XDhslPgAXjgilCM4JJzmK2cCwREvKRk-JTUtXQMAqDNy3IA6C05bFzcOiS7fBV6NbT1DE2HeUeYJkKmIxxcYoA)

@veronikaEve is your use case represented in one of these? If not, could you share with us a TypeScript Playground reproducing your use case?

Thank you!

**Reply by veronikaEve** [reply]
I just tried, and the scenarios in your typescript playground are exactly what i can see on my end too! So the only version that works for me in typescript is

```
import { request } from '@octokit/request';
request(("GET"))
```

Good to know something does work!

Also I I re-tried in a .js file and 
```
import {Octokit} from "octokit";
const octokit = new Octokit();

octokit.request(("GET"))
```
does actually work in .js, just not in .ts

**Reply by veronikaEve** [reply]
I guess I do have more questions now in regards to other implementation. 

https://github.com/octokit/oauth-app.js/#for-oauth-apps
The snippet shown here as an example uses `octokit.request()`. 
How could I refactor this to use the standalone request? Or is my only option now to not use typescript?
